src/sys/socket/sockopt.rs

```diff
@@ -1115,6 +1115,158 @@ where
     }
 }
 
+/// Set the Upper Layer Protocol (ULP) on the TCP socket.
+///
+/// For example, to enable the TLS ULP on a socket, the C function call would be:
+///
+/// ```c
+/// setsockopt(sock, SOL_TCP, TCP_ULP, "tls", sizeof("tls"));
+/// ```
+///
+/// ... and the `nix` equivalent is:
+///
+/// ```ignore,rust
+/// setsockopt(sock, TcpUlp::default(), b"tls\0");
```

Comment:
I am curious, in the C interface, does this "tls" argument always have the type C string (having a NUL at the end) or it can be some pure random bytes

Reply:
The value is the name of the ULP. You can look at the ones you have available with `cat /proc/sys/net/ipv4/tcp_available_ulp`.

The string does not need to be NUL-terminated. This is leftover from when I was trying to make the CI work and I thought the lack of NUL terminator might be the reason armv7 was failing. I confirmed from the kernel code that it only copies the user-provided string up to the smaller of user-provided length and 15 bytes (`TCP_ULP_NAME_MAX - 1`) and adds its own NUL terminator, and it also works without the trailing NUL in my test program.

Reply:
Removed the trailing NUL from this example and from the test.

Reply:
>  I confirmed from the kernel code that it only copies the user-provided string up to the smaller of user-provided length and 15 bytes (TCP_ULP_NAME_MAX - 1) and adds its own NUL terminator.

Thanks for digging into the kernel code, do you think it would be great if we could have a comment in the code describing this behavior?

Reply:
Done.

Reply:
Thanks for doing it, and you don't need to squash commits as we will do squash merge.